src/python/pants/backend/native/tasks/native_third_party_fetch.py

```diff
@@ -0,0 +1,242 @@
+
+  def execute(self):
+    native_lib_tgts = self.context.targets(self.native_library_constraint.satisfied_by)
+    with self.invalidated(native_lib_tgts,
+                          invalidate_dependents=True) as invalidation_check:
+      for vt in invalidation_check.all_vts:
+        if vt.valid:
+          self.populate_task_product(vt)
+        else:
+          self.fetch_packages(vt)
+
+  def populate_task_product(self, vt):
+    task_product = {}
+    task_product['lib_names'] = []
+
+    lib = os.path.join(vt.results_dir, 'lib')
+    include = os.path.join(vt.results_dir, 'include')
+
+    if os.path.exists(lib):
+      task_product['lib'] = lib
+      for filename in os.listdir(lib):
+        lib_name = self.parse_lib_name_from_library_filename(filename)
+        if lib_name:
+          task_product['lib_names'].append(lib_name)
+
+    if os.path.exists(include):
+      task_product['include'] = include
+
+    self.context.products.register_data(self.ThirdPartyLibraryFiles, task_product)
+
+  def ensure_conan_remote_configuration(self, conan_binary):
+    """
+    Ensure that the conan registry.txt file is sanitized and loaded with
+    a pants-specific remote for package fetching.
+
+    :param conan_binary: The conan client pex to use for manipulating registry.txt.
+    """
+
+    # Delete the conan-center remote from conan's registry.
+    remove_conan_center_remote_cmdline = conan_binary.pex.cmdline(['remote',
+                                                                   'remove',
+                                                                   'conan-center'])
+    try:
+      stdout = subprocess.check_output(remove_conan_center_remote_cmdline.split())
+      self.context.log.debug(stdout)
+    except subprocess.CalledProcessError as e:
+      if not "'conan-center' not found in remotes" in e.output:
+        raise TaskError('Error deleting conan-center from conan registry: {}'.format(e.output))
+
+    # Add the pants-specific conan remote.
+    remote_url = self.get_options().conan_remote
+    add_pants_conan_remote_cmdline = conan_binary.pex.cmdline(['remote',
+                                                               'add',
+                                                               'pants-conan-remote',
+                                                               remote_url,
+                                                               '--insert'])
+    try:
+      stdout = subprocess.check_output(add_pants_conan_remote_cmdline.split())
+      self.context.log.debug(stdout)
+    except subprocess.CalledProcessError as e:
+      if not "Remote 'pants-conan-remote' already exists in remotes" in e.output:
+        raise TaskError('Error adding pants-specific conan remote: {}'.format(e.output))
+
+  def copy_package_contents_from_conan_dir(self, results_dir, conan_requirement, pkg_sha):
+    """
+    Copy the contents of the fetched pacakge into the results directory of the versioned
```

Comment:
Typo: s/pacakge/package/

Can we symlink instead of copying? That's what we do for ivy-resolved jars (iirc they are symlinks from the .ivy2 cache). Or possibly that's the wrong thing to do? Either way, we should be consistent?

Reply:
It would be consistent with Coursier and Ivy, so I will look into supporting that.

Reply:
After further investigation, this may not be feasible for the current iteration and would probably be better for a follow up ticket.

The reason is that the fetch task aggregate the contents of subdirectories of each conan package into two "product" folders: bin/ and lib/. AFAICT, this would require symlinking recursively to every file in the dirs of each conan package from the data dir, and relying on the symlinking command to de-dupe. Based on my attempts to do this, I could not come up with a simple and effective solution. I am happy to hear suggestions though, but it may make sense to punt on this to unblock in the short term.